src/main/java/jrpc/clightning/plugins/CLightningPlugin.java

```diff
@@ -158,15 +159,22 @@ public void start() {
           // TODO refactoring this inside the reflectionManager and add a list of methods inside
           // this class
           CLightningJsonObject notificationRequest = new CLightningJsonObject(object);
-          for (Method method : reflections.getMethodsAnnotatedWith(Subscription.class)) {
-            if (method.isAnnotationPresent(Subscription.class)) {
-              Subscription subscription = method.getAnnotation(Subscription.class);
-              String event = subscription.notification();
-              String methodName = notificationRequest.get("method").getAsString();
-              if (methodName.equals(event)) {
-                method.invoke(this, notificationRequest);
-              }
-            }
+
+          List<Method> eligibleMethods =
+              reflections.getMethodsAnnotatedWith(Subscription.class).stream()
+                  .filter(it -> it.isAnnotationPresent(Subscription.class))
+                  .filter(it -> Arrays.asList(this.getClass().getDeclaredMethods()).contains(it))
+                  .filter(
+                      it -> {
+                        Subscription subscription = it.getAnnotation(Subscription.class);
+                        String event = subscription.notification();
+                        String methodName = notificationRequest.get("method").getAsString();
+                        return methodName.equals(event);
+                      })
+                  .collect(Collectors.toList());
+
+          for (Method method : eligibleMethods) {
+            method.invoke(this, notificationRequest);
```

Comment:
Now that we are rewriting this code maybe we can fix my previous TODO by moving this search code inside a function as I did for other annotation?

Reply:
If that is okay for you, I'll do that in a follow-up PR. :pray:

Reply:
it is perfect :)